src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +213,32 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount > 2)
```

Comment:
Then I don't really get what you're trying to say either.

I was talking about this implementation, which currently allows you to set a channel count of 0, which is undefined and will do who knows what. :wink:
If you want to suggest a different design all together you should mark that clearly and/or not directly reply to my point that talks about the current implementation.